commit e0a033b1762b5f0d93b3ac21bd003749c5fb2533
Author: Tobias Klauser <tklauser@distanz.ch>
Date:   Thu Jan 12 16:53:11 2017 +0100

    x86/mpx: Use compatible types in comparison to fix sparse error
    
    
    [ Upstream commit 453828625731d0ba7218242ef6ec88f59408f368 ]
    
    info->si_addr is of type void __user *, so it should be compared against
    something from the same address space.
    
    This fixes the following sparse error:
    
      arch/x86/mm/mpx.c:296:27: error: incompatible types in comparison expression (different address spaces)
    
    Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
    Cc: Dave Hansen <dave.hansen@linux.intel.com>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: linux-kernel@vger.kernel.org
    Signed-off-by: Ingo Molnar <mingo@kernel.org>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/x86/mm/mpx.c b/arch/x86/mm/mpx.c
index e4f8009..3e7c489 100644
--- a/arch/x86/mm/mpx.c
+++ b/arch/x86/mm/mpx.c
@@ -293,7 +293,7 @@ siginfo_t *mpx_generate_siginfo(struct pt_regs *regs)
 	 * We were not able to extract an address from the instruction,
 	 * probably because there was something invalid in it.
 	 */
-	if (info->si_addr == (void *)-1) {
+	if (info->si_addr == (void __user *)-1) {
 		err = -EINVAL;
 		goto err_out;
 	}
